service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java

```diff
@@ -164,6 +179,103 @@ private void processCredentialsMessageGetOperation(final Message<JsonObject> reg
         getCredentials(tenantId, type, authId, result -> reply(regMsg, result));
     }
 
+    private void processCredentialsMessageAddOperation(final Message<JsonObject> regMsg, final String tenantId, final JsonObject payload) {
+        if (!isValidCredentialsObject(payload)) {
+            reply(regMsg, CredentialsResult.from(HTTP_BAD_REQUEST));
+            return;
+        }
+        addCredentials(tenantId, payload, result -> reply(regMsg, result));
+    }
+    
+    private void processCredentialsMessageUpdateOperation(final Message<JsonObject> regMsg, final String tenantId, final JsonObject payload) {
+        if (!isValidCredentialsObject(payload)) {
```

Comment:
This does not currently check if there was an empty json array for the secrets, does it?
But an update operation with no secrets contained must be avoided IMHO.

Reply:
... see previous question